### ISSUE TITLE
::CompositeIO & ::Parts deprecation warnings

### ISSUE THREAD

**Issue by samcolson4** [cloudfoundry/cloud_controller_ng#2985]
## Issue

When running unit tests, several deprecation warnings appear:

```
Top level ::CompositeIO is deprecated, require 'multipart/post' and use `Multipart::Post::CompositeReadIO` instead!
Top level ::Parts is deprecated, require 'multipart/post' and use `Multipart::Post::Parts` instead!
```

## Steps to Reproduce

Run unit tests:
`bundle exec rake spec:all spec/unit/middleware/rate_limiter_v2_api_spec.rb`

## Expected result
No deprecation warnings should be shown.

## Current result
Deprecation warnings are shown, as above.

## Possible Fix
The `faraday-multipart` dependency on `multipart-post` is known to be a common cause. This gem [has been updated](https://github.com/sul-dlss/argo/pull/3732) to a new version which could be used.

**Comment by will-gant** [comment]
Had a bit of a delve into this, but haven't had a chance to try fixes yet.

The two warnings are emitted ([here](https://github.com/socketry/multipart-post/blob/main/lib/composite_io.rb) and [here](https://github.com/socketry/multipart-post/blob/main/lib/parts.rb)) by versions above 2.2.0 of the gem `multipart-post`. We're [consuming](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/Gemfile.lock#L327) v2.2.3.

 `multipart-post` is a [child dependency](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/Gemfile.lock#L153-L154) of v0.17.5 of `faraday`, and it looks to me like the first version of that which won't suffer from this is going to be v2.0.0, which actually [removed](https://github.com/lostisland/faraday/commit/cb47eca810c15512c91379f0d0a0fd9dcf143566) `multipart-post` entirely as a dependency. This is roughly how we're consuming `multipart-post` at the moment:

- `multipart-post` 2.2.3
    - `faraday` 0.17.5
        - `faraday_middleware` 0.14.0
            - `azure-core` 0.1.15
                - `azure-stoage` ([fork](https://github.com/sethboyles/azure-storage-ruby/tree/x-ms-blob-content-type-fix-1.1) based on v1.1.0)
                    - `fog-azure-rm` (a special branch for CF: [`fog-arm-cf`](https://github.com/fog/fog-azure-rm/tree/fog-arm-cf))

**Comment by philippthun** [comment]
Fixed with #3259.
